app/script/announce/AnnounceRepository.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Wire
+ * Copyright (C) 2016 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+(function() {
+  window.z = window.z || {};
+  window.z.announce = z.announce || {};
+
+  const ANNOUNCE_CONFIG = {
+    CHECK_INTERVAL: 3 * 60 * 60 * 1000,
+    CHECK_TIMEOUT: 5 * 60 * 1000,
+    UPDATE_INTERVAL: 6 * 60 * 60 * 1000,
+  };
+
+  window.z.announce.AnnounceRepository = class AnnounceRepository {
+    constructor(announce_service) {
+      this.logger = new z.util.Logger('z.announce.AnnounceRepository', z.config.LOGGER.OPTIONS);
+      this.announce_service = announce_service;
+      return this;
```

Comment:
I removed the `this`. It was there because otherwise CoffeeScript would return the last statement from the constructor (instead the instance of the class).